packages/kbn-es/src/cluster.js

```diff
@@ -0,0 +1,131 @@
+const childProcess = require('child_process');
+const chalk = require('chalk');
+const { installSnapshot, installSource, installArchive } = require('./install');
+const { ES_BIN } = require('./paths');
+const { log, parseEsLog, extractConfigFiles } = require('./utils');
+
+exports.Cluster = class Cluster {
+  /**
+   * Builds and installs ES from source
+   *
+   * @param {Object} options
+   * @property {Array} options.installPath
+   * @property {Array} options.sourcePath
+   * @returns {Promise}
+   */
+  async installSource(options = {}) {
+    log.info(chalk.bold('Installing from source'));
+    log.indent(2);
```

Comment:
I personally think 4 space indentation looks better in the tooling log, but that's just me :)